src/guillotine/fetchContent.ts

```diff
@@ -395,27 +395,46 @@ function prefixLayoutPath(contentType: string, path: string): string {
 
 function buildPage(contentType: string, comps: PageComponent[] = []): PageComponent {
 
-    let page: PageComponent = {
+    const page: PageComponent = {
         type: XP_COMPONENT_TYPE.PAGE,
         path: '/',
     };
-    const tree = {};
-    if (contentType === FRAGMENT_CONTENTTYPE_NAME) {
-        page.regions = tree;
+
+    return buildComponentTree(comps, page, contentType);
+}
+
+function buildComponentTree(comps: PageComponent[], rootComp: PageComponent, contentType?: string): PageComponent {
+    const rootTree: RegionTree = {};
+    const isFragmentType = contentType === FRAGMENT_CONTENTTYPE_NAME;
+
+    if (isFragmentType) {
+        rootComp.regions = rootTree;
     }
 
     comps.forEach(cmp => {
         const cmpPath = parseComponentPath(contentType, cmp.path);
-        let region;
-        if (cmp.path === '/' && cmp.type === XP_COMPONENT_TYPE.PAGE) {
-            // add page values to page object
-            page = Object.assign(page, cmp);
-            page.page.regions = tree;
-            return;
-        } else {
-            region = getParentRegion(tree, contentType, cmpPath, comps, true);
+
+        if (cmp.path === '/') {
+            if (cmp.type === XP_COMPONENT_TYPE.PAGE && rootComp.type === XP_COMPONENT_TYPE.PAGE) {
+                // add page values to page object
+                rootComp = Object.assign(rootComp, cmp);
+                rootComp.page.regions = rootTree;
+                return;
+            } else if (cmp.type === XP_COMPONENT_TYPE.LAYOUT && rootComp.type === XP_COMPONENT_TYPE.FRAGMENT) {
+                // when rendering fragment as part of the page with layout at root path
+                // remove other children from list as they will be included in layout
+                rootComp.fragment.fragment.components = [cmp];
+                cmp.regions = rootTree;
+            }
```

Comment:
What is going to happen if I have a fragment at the top and other components too?
/main/0/ <- fragment 
/main/1/ <- layout
/main/2/ <- part

Reply:
This function is recursive, so this clause is executed when we're inside of a fragment because of `rootComp.type === XP_COMPONENT_TYPE.FRAGMENT` and there is a layout at the root path, it's not applied for regular page layouts